### PR TITLE
[incubator/cni-genie] Addition of chart for cni genie deployment

### PR DESCRIPTION
**Pull request by swx457056**
#### What this PR does / why we need it:
This PR includes the chart for CNI Genie deployment on kubernetes. It is plugin which does multi-networking provisioning for pods. Github link (https://github.com/cni-genie/CNI-Genie)

#### Which issue this PR fixes
First PR to add helm chart for cni genie

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] title of the PR contains starts with chart name e.g. `[stable/chart]`

[helm-CNI-Genie-TestReport.docx](https://github.com/helm/charts/files/3176818/helm-CNI-Genie-TestReport.docx)

### CODE REVIEW

File: incubator/cni-genie/templates/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (mattfarina) [inline incubator/cni-genie/templates/clusterrole.yaml]: For RBAC can you please use the `apiVersion` of `rbac.authorization.k8s.io/v1`. RBAC has been GA for quite a few versions now. This applies to all the objects.

File: incubator/cni-genie/templates/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
+# Install CNI-Genie plugin on each slave node.
+kind: DaemonSet
+apiVersion: extensions/v1beta1
```
Comment (mattfarina) [inline incubator/cni-genie/templates/daemonset.yaml]: Can you please use `apps/v1` for the `apiVersion`. The extensions one is deprecated and soon k8s will stop accepting them. This applies to all the daemonsets.

File: incubator/cni-genie/values.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+image:
+  pullPolicy: IfNotPresent
+
+  # Paths for fetching genie images
+  pluginImagePath: quay.io/huawei-cni-genie/genie-plugin:latest
+
+  admissionControllerImagePath: quay.io/huawei-cni-genie/genie-admission-controller:latest
+
+  policyControllerImagePath: quay.io/huawei-cni-genie/genie-policy-controller:latest
```
Comment (mattfarina) [inline incubator/cni-genie/values.yaml]: For the images can you please follow the format [documented in the review docs](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#configuration).

File: incubator/cni-genie/templates/service.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "cni-genie.fullname" . }}
+  labels:
+    app: {{ template "cni-genie.name" . }}
+    chart: {{ template "cni-genie.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
```
Comment (mattfarina) [inline incubator/cni-genie/templates/service.yaml]: Can you please use the newer format for the [labels in the docs](https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#metadata). These come from a common set documented by Kubernetes. This applies all over.

**Review by mattfarina — COMMENTED** [review]
Thanks for the PR. There are some minor changes we need in the chart.

**Comment by sushanthakumar** [comment]
@swx457056 , Pls make sure to address these aspects in all relevant places/files, thanks

File: incubator/cni-genie/values.yaml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  policy:
+    create: true
+    name:
+
+# Names for genie plugin and policy
+plugin:
+  name: genie-plugin
+  image:
+    pullpolicy: IfNotPresent
+    tag: latest
```
Comment (mattfarina) [inline incubator/cni-genie/values.yaml]: Can you use image tags for stable versions. See https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#configuration

It idea is to have repeatable releases. If one deploys this chart into QA and then into production the exact same images should be used. `latest` can move which breaks the repeatability.

This applies to all images.

**Comment by maorfr** [comment]
please take a look at https://github.com/helm/helm/pull/6095#issue-301608768

we strongly encourage people to self host chart repos:
https://github.com/helm/hub/blob/master/Repositories.md

you can even do that using GitHub pages:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

**Pull request closed without merging**